pkgs/by-name/do/dorion/package.nix

```
@@ -1,65 +1,182 @@
-{ lib
-, stdenv
-, fetchurl
-, autoPatchelfHook
-, dpkg
-, glib-networking
-, gst_all_1
-, libappindicator
-, libayatana-appindicator
-, webkitgtk_4_0
-, wrapGAppsHook3
+{
+  lib,
+  fetchFromGitHub,
+  fetchurl,
+  rustPlatform,
+  cmake,
+  wrapGAppsHook3,
+  glib-networking,
+  gst_all_1,
+  libsysprof-capture,
+  libayatana-appindicator,
+  nodejs,
+  openssl,
+  pkg-config,
+  yq-go,
+  pnpm_9,
+  webkitgtk_4_1,
+  cargo-tauri,
+  desktop-file-utils,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
-  pname = "dorion";
-  version = "5.0.1";
+let
+  webkitgtk_4_1' = webkitgtk_4_1.override {
+    enableExperimental = true;
+  };
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-cCZikTZ+IU3mq/FkJfeggXLyWIsWG+a2qu1GbgW93WQ=";
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/de7151650c240fc226d41ff898547edc4d23c94d/shelter.js";
+    hash = "sha256-dqAbq4O61t91gnVBOiN5KgJG3Frg9ePMPKjn9QWtkws=";
+    meta = {
+      homepage = "https://github.com/uwu/shelter";
+      sourceProvenance = [ lib.sourceTypes.binaryBytecode ]; # actually, minified JS
+      license = lib.licenses.cc0;
+    };
   };
+in
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
 
-  runtimeDependencies = [
-    glib-networking
-    libappindicator
-    libayatana-appindicator
+# nyo finalAttrs :<
+# https://github.com/NixOS/nixpkgs/pull/194475
+rustPlatform.buildRustPackage rec {
+  pname = "dorion";
+  version = "6.4.1";
+
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = "Dorion";
+    rev = "v${version}";
+    hash = "sha256-La58qj9acZzTG7WDfnBNdVWzAjOYVkIZzO8uyjQvt6o=";
+  };
+
+  cargoPatches = [
+    ./cargo.patch
   ];
 
+  cargoRoot = "src-tauri";
+  buildAndTestSubdir = cargoRoot;
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-5qK1IRaUrf4pYDmoBZoR6q6rL06irurwo4v5JLX+Io4=";
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit pname version src;
+    hash = "sha256-xBonUzA4+1zbViEsKap6CaG6ZRldW1LjNYIB+FmVRFs=";
+  };
+
   nativeBuildInputs = [
-    autoPatchelfHook
-    dpkg
+    pnpm_9.configHook
+    rustPlatform.cargoSetupHook
+    cargo-tauri.hook
+    rustPlatform.cargoCheckHook
```

```suggestion
    cargo-tauri.hook
```
These are already brought in with `buildRustPackage`